pkg/util/azureclient/authz/remotepdp/client.go

```diff
@@ -0,0 +1,91 @@
+package remotepdp
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+var _ RemotePDPClient = &remotePDPClient{}
+
+// RemotePDPClient represents the Microsoft Remote PDP API Spec
+type RemotePDPClient interface {
+	CheckAccess(context.Context, AuthorizationRequest) (*AuthorizationDecisionResponse, error)
+}
```

Comment:
can you add a little comment explaining why we have that line ? (16) (it provides compile time assertion that our type is implementing the interface, but it's not necessarily obvious to every one)

Reply:
Added as suggested.